eventuate-core/src/test/scala/com/rbmhtechnology/eventuate/EventsourcedActorTestKitSpec.scala

```diff
@@ -0,0 +1,320 @@
+/*
+ * Copyright 2015 - 2016 Red Bull Media House GmbH <http://www.redbullmediahouse.com> - all rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.rbmhtechnology.eventuate
+
+import akka.actor._
+import akka.pattern.ask
+import akka.testkit._
+import com.rbmhtechnology.eventuate._
+import org.scalatest._
+import org.scalatest.exceptions.TestFailedException
+
+import scala.util._
+
+case class Ping(i: Int)
+
+case class Pong(i: Int)
+
+case class Request(s: String)
+
+case class RequestWithPersistence(s: String, persistenceFirst: Boolean)
+
+case class Response(s: String)
+
+case class PersistSlowly(remaining: Seq[Int])
+
+class SimpleActor(log: ActorRef) extends EventsourcedActor with PersistOnEvent {
+
+  override def id: String = "simple-actor"
+
+  override def eventLog: ActorRef = log
+
+  var secretState = 0
+
+  override def onEvent: Receive = {
+    case p: Pong =>
+      secretState = p.i
+
+    case Ping(i) =>
+      persistOnEvent(Pong(i))
+  }
+
+  override def onCommand: Receive = {
+    case Ping(i) =>
+      persist(Pong(i)) {
+        case Success(e)  =>
+        case Failure(ex) =>
+      }
+
+    case Request(request) =>
+      sender ! Response(request)
+
+    case RequestWithPersistence(request, false) =>
+      sender ! Response(request)
+      persist(Response(request)) {
+        case Success(e)  =>
+        case Failure(ex) =>
+      }
+
+    case RequestWithPersistence(request, true) =>
+      persist(Response(request)) {
+        case Success(e)  =>
+        case Failure(ex) =>
+      }
+      sender ! Response(request)
```

Comment:
This sends the response **concurrently** to `persist`, not afterwards, because `persist` is an **asynchronous** operation. In order to send a response after successful persistence, this must be done in the success case of the persist handler.